backend/api/models.py

```
@@ -218,25 +218,30 @@ def get_preference_based_videos(cls, user, preferences, limit, offset=0):
         return cls.objects.filter(
             visibility='public'
         ).annotate(
             popularity_score=ExpressionWrapper(
                 (F('views') * 0.6) + (F('likes') * 0.4),
                 output_field=FloatField()
             )
-        ).order_by('-popularity_score')[offset:offset+limit]
+        ).order_by('-popularity_score')
+
+    @classmethod
+    def get_popular_videos(cls, limit, offset=0):
```

Ruff (0.8.2)

Missing return type annotation for classmethod `get_popular_videos`

(ANN206)